### ISSUE TITLE
field type is changed when using latest userforms version

### ISSUE THREAD

**Issue by guyvanbael** [silverstripe/silverstripe-mathspamprotection#32]
When i updated to the latest userforms module, the EditableMathSpamField was replaced by an EditableTextField and form can't be submitted anymore.

**Comment by Taitava** [comment]
I can confirm this. I'll try to see what's wrong and will get back if I find anything. In the mean time, if anybody has any clue about this, please tell :). I can't promise any solution yet.

**Comment by tractorcow** [comment]
Is it a server side or client side validation which is blocking submission?

**Comment by Taitava** [comment]
@tractorcow It's server side. I have real time validation disabled.

**Comment by tractorcow** [comment]
I remember fixing userforms a while back to work with the spam protector; I'll check if the code still works. ;)

You are using the EditableSpamProtector field, right?

**Comment by guyvanbael** [comment]
yes, i am.

**Comment by Taitava** [comment]
Me too.

**Comment by tazzydemon** [comment]
The way validation is handled appears to have changed enormously on Userforms 3.x and as far as I can tell has broken all the standard spamprotection modules including mine which was an upgrade of the old honeypot. I have asked elsewhere for the api instructions or description for how this has changed?

**Comment by guyvanbael** [comment]
Thanks for looking into it. Meanwhile i'm using the invisiblespamprotection module, but the mathspamprotection always did a great job!

**Comment by tazzydemon** [comment]
Are you saying that invisiblespamprotection works for you in Userforms 3.x? It doesn't for me

**Comment by guyvanbael** [comment]
Yes it does, but it isn't half as effective as mathspamprotection

**Comment by tazzydemon** [comment]
I cant find any evidence that it works with userforms 3. I can fill in the honeypot on any of the various versions including mine and the form does not get returned but submitted. Curious

**Comment by tazzydemon** [comment]
The problem was my selected version of silverstripe-spamprotection which has to be 2.0.2+ (or master) 
There was a key commit made here:
https://github.com/silverstripe/silverstripe-spamprotection/commit/da87f71a158905b4732aa21bc20c4833260a7dfb

**Comment by guyvanbael** [comment]
So did you get mathspamprotection working again? These are my entries in my composer file, but it doesn't seem to work.
"silverstripe/spamprotection": "*",
"silverstripe/userforms": "dev-master",
"silverstripe/mathspamprotection":"*",

**Comment by tazzydemon** [comment]
You missed this line in composer.json
        "silverstripe/spamprotection": "~2.0.2",

You also need a spamprotection.yml in your _config directory with this in it:

---
name: spamprotection
---
FormSpamProtectionExtension:
#  default_spam_protector:
  default_spam_protector:  MathSpamProtector

**Comment by tazzydemon** [comment]
By the way I have found the problem and will post an issue fix

**Comment by Taitava** [comment]
@tazzydemon Hi, have you made any progress with the fix? Or can you provide further information about the problem? Thanks a lot! :)

**Comment by tazzydemon** [comment]
Because I'm not using this module I didn't fork and pull request like I did with silverstripe-spamprotection but if you check out issue 34 you will see the fix https://github.com/silverstripe/silverstripe-mathspamprotection/issues/42
